ext/intl/rangeformatter/rangeformatter_class.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Bogdan Ungureanu <bogdanungureanu21@gmail.com>              |
+   +----------------------------------------------------------------------+
+*/
+
+#include <unicode/numberrangeformatter.h>
+#include <unicode/unumberrangeformatter.h>
+#include <unicode/numberformatter.h>
+#include <unicode/unistr.h>
+#include "../intl_convertcpp.h"
+
+extern "C" {
+    #include "php.h"
+    #include "../php_intl.h"
+    #include "../intl_data.h"
+    #include "rangeformatter_arginfo.h"
+    #include "rangeformatter_class.h"
+#include "intl_convert.h"
+}
+
+using icu::number::NumberRangeFormatter;
+using icu::number::NumberFormatter;
+using icu::number::UnlocalizedNumberFormatter;
+using icu::number::LocalizedNumberRangeFormatter;
+using icu::UnicodeString;
+using icu::MeasureUnit;
+
+static zend_object_handlers rangeformatter_handlers;
+zend_class_entry *class_entry_IntlNumberRangeFormatter;
+
+zend_object *IntlNumberRangeFormatter_object_create(zend_class_entry *ce)
+{
+	IntlNumberRangeFormatter_object* intern;
+
+	intern = (IntlNumberRangeFormatter_object*)zend_object_alloc(sizeof(IntlNumberRangeFormatter_object), ce);
+	zend_object_std_init(&intern->zo, ce);
+	object_properties_init(&intern->zo, ce);
+
+	return &intern->zo;
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, __construct)
+{
+   
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, createFromSkeleton)
+{
+    char* skeleton;
+    char* locale;
+    size_t locale_len;
+    size_t skeleton_len;
+    zend_long collapse;
+    zend_long identityFallback;
+
+    ZEND_PARSE_PARAMETERS_START(4,4)
+        Z_PARAM_STRING(skeleton, skeleton_len)
+        Z_PARAM_STRING(locale, locale_len)
+        Z_PARAM_LONG(collapse)
+        Z_PARAM_LONG(identityFallback)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if (locale_len == 0) {
+        locale = (char *)intl_locale_get_default();
+    }
+
+    if (skeleton_len == 0) {
+        zend_argument_value_error(1, "Skeleton string cannot be empty");
```

Comment:
The error messages don't fit `zend_argument_value_error`. E.g. the result of this one will read as `Argument #1 ($skeleton) Skeleton string cannot be empty`. So you should change the 2nd argument to `cannot be empty`.

Reply:
Better yet, use `zend_argument_must_not_be_empty_error(1);` which will ensure standard wording